cmd/evm/internal/t8ntool/block.go

```diff
@@ -156,7 +156,7 @@ func (i *bbInput) ToBlock() *types.Block {
 	if i.Header.Nonce != nil {
 		header.Nonce = *i.Header.Nonce
 	}
-	if header.Difficulty != nil {
+	if i.Header.Difficulty != nil {
```

Comment:
nice catch, looks like this mistake was copied from upstream: https://github.com/ethereum/go-ethereum/blob/master/cmd/evm/internal/t8ntool/block.go#L161

Do you want to make a PR to geth to get credit for your work upstream as well?

Reply:
Hah, I didn't even think of that. Will send them a PR too, thanks.